src/dataset.hpp

```diff
@@ -62,7 +62,13 @@ struct randomx_cache {
 //A pointer to a standard-layout struct object points to its initial member
 static_assert(std::is_standard_layout<randomx_dataset>(), "randomx_dataset must be a standard-layout struct");
 //the following assert fails when compiling Debug in Visual Studio (JIT mode will crash in Debug)
+#if _MSC_VER && !__INTEL_COMPILER && _DEBUG
+#define _STR(x) #x
+#define STR(x) _STR(x)
+#pragma message ( __FILE__ "(" STR(__LINE__) ") warning: check std::is_standard_layout<randomx_cache>() is disabled for Debug configuration" )
```

Comment:
This assert must be there because ASM code assumes that pointer to cache memory is the first member of randomx_cache struct, this is exactly what std::is_standard_layout checks.

Reply:
Done (see comments below)